src/common/cups_print.c

```diff
@@ -323,61 +398,120 @@ void dt_print_file(const int32_t imgid, const char *filename, const dt_print_inf
     return;
   }
 
-  cups_dest_t *dests;
-  int num_dests = cupsGetDests(&dests);
-  cups_dest_t *dest = cupsGetDest(pinfo->printer.name, NULL, num_dests, dests);
-
   cups_option_t *options = NULL;
   int num_options = 0;
 
-  for (int j = 0; j < dest->num_options; j ++)
-    if (cupsGetOption(dest->options[j].name, num_options,
-                      options) == NULL)
-      num_options = cupsAddOption(dest->options[j].name,
-                                  dest->options[j].value,
-                                  num_options, &options);
-
-  cupsFreeDests(num_dests, dests);
+  // for turboprint drived printer, use the turboprint dialog
+  if (pinfo->printer.is_turboprint)
+  {
+    const char *tp_intent_name[] = { "perception_0", "colorimetric-relative_1", "saturation_1", "colorimetric-absolute_1" };
+    char tpcmd[1024];
+    char tmpfile[100];
 
-  // if we have a profile, disable cm on CUPS, this is important as dt does the cm
+    snprintf(tmpfile, sizeof(tmpfile), "/tmp/dt_cups_opts_%d", imgid);
```

Comment:
Please use `g_mkstemp()` or `g_file_open_tmp()` instead of trying to come up with a unique name. This is opening us up for race conditions.